### DISCUSSION TITLE
Filter the commit history by file paths and/or commit msg?

### DISCUSSION THREAD

**Discussion by BinToss** [GitTools/GitVersion#2770] [Ideas]
CommitsSinceVersionSource is fine as-is, but I'd prefer BuildMetaData to reflect the number of commits or pushes that affected the compiled assemblies. This would require additional configuration on the users' end to set up and would default to the current behavior. 

This could be done automatically for some projects by comparing a list of changed files to the list of files in the Microsoft .*proj files. Then, if the former lacks all files from the latter, skip GitVersion operation. Additional configuration may be necessary for edge cases that use pre-build or post-build scripts to affect the build artifacts.

#### Example A
If a CI build is cancelled or skipped, but didn't explicitly fail, the next successful build should reuse the version info.
#### Example B
In the previous example, the build may have been automatically cancelled by a script or set of rules that determine if the build artifacts would be significantly different from the previous build. I use something like that to determine if a GitHub Release should be created or not. My use-case determines it based on if the Push does not include changes to `./src/`, including/excluding paths e.g. include gitmodule build dependency, exclude nested ReadMe files.

**Comment by asbjornu** [comment]
Filtering on file path sounds like a reasonable addition to the filtering capabilities. PRs are welcome.

**Reply by BinToss** [reply]
The soonest I can get around to this is in a few months, after I get some higher priority projects to acceptable stopping points.

**Reply by HHobeck** [reply]
You should take a look to this feature implementation initiative: [filter on tags on prefix or regex, to make it possible to use multiple git versions in one repository #2436](https://github.com/GitTools/GitVersion/issues/2436)
